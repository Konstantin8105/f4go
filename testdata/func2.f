       FUNCTION func_name(a, b)
          INTEGER :: func_name
          INTEGER :: a
          REAL    :: b
          func_name = (2*a)+b
          RETURN
       END FUNCTION
   
       PROGRAM cows
          IMPLICIT NONE
          INTEGER :: func_name
          PRINT *,func_name(2, 1.3)
       END PROGRAM
