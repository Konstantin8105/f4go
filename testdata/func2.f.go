package main

func func_name() int {
	{
		var __result_func_name int
		__result_func_name = *a*2 + *b
		return f4goUndefinedmodify_expr
		return f4goUndefinedmodify_expr
	}
}
func cows() {
	{
		{
			var dt_parm.0 __st_parameter_dt
			dt_parm.0.common.filename = "testdata/func2.f "[1]
			dt_parm.0.common.line = 12
			dt_parm.0.common.flags = 128
			dt_parm.0.common.unit = 6
			_gfortran_st_write(dt_parm.0)
			{
				var tempVarF4GO_42 int
				tempVarF4GO_42 = func_name(f4goUndefinedGetNameFunctionconst_decl, f4goUndefinedGetNameFunctionconst_decl)
				_gfortran_transfer_integer_write(dt_parm.0, tempVarF4GO_42, 4)
			}
			_gfortran_st_write_done(dt_parm.0)
		}
	}
}
func main() int {
	{
		var options.1 [288]int
		_gfortran_set_args(argc, argv)
		_gfortran_set_options(9, options.1[0])
		cows()
		return f4goUndefinedmodify_expr
	}
}
