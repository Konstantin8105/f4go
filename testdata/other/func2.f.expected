FUNCTION            	FUNCTION
IDENT               	func_name
(                   	(
IDENT               	a
,                   	,
IDENT               	b
)                   	)
NEW_LINE            	

INTEGER             	INTEGER
DOUBLE_COLON        	::
IDENT               	func_name
NEW_LINE            	

INTEGER             	INTEGER
DOUBLE_COLON        	::
IDENT               	a
NEW_LINE            	

IDENT               	REAL
DOUBLE_COLON        	::
IDENT               	b
NEW_LINE            	

IDENT               	func_name
=                   	=
(                   	(
INT                 	2
*                   	*
IDENT               	a
)                   	)
+                   	+
IDENT               	b
NEW_LINE            	

return              	RETURN
NEW_LINE            	

END                 	END
FUNCTION            	FUNCTION
NEW_LINE            	

NEW_LINE            	

PROGRAM             	PROGRAM
IDENT               	cows
NEW_LINE            	

IMPLICIT            	IMPLICIT
IDENT               	NONE
NEW_LINE            	

INTEGER             	INTEGER
DOUBLE_COLON        	::
IDENT               	func_name
NEW_LINE            	

IDENT               	PRINT
*                   	*
,                   	,
IDENT               	func_name
(                   	(
INT                 	2
,                   	,
FLOAT               	1.3
)                   	)
NEW_LINE            	

END                 	END
PROGRAM             	PROGRAM
NEW_LINE            	

